sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -593,9 +594,18 @@ def __init__(
     Args:
       task (str or enum.Enum): task supported by HuggingFace Pipelines.
         Accepts a string task or an enum.Enum from PipelineTask.
-      model : path to pretrained model on Hugging Face Models Hub to use custom
-        model for the chosen task. If the model already defines the task then
-        no need to specify the task parameter.
+      model (str): path to the pretrained *model-id* on Hugging Face Models Hub
```

Comment:
Do we intentionally  use `model_uri` in other places but `model` here? 

https://github.com/apache/beam/blob/0f727927c0136c63e8127ab5e9d3279fd5748dc4/sdks/python/apache_beam/ml/inference/huggingface_inference.py#L421

Reply:
if this param is not a uri i am confused what 'path' would look like here besides the id itself.

Reply:
That's right. This is consistent with the hugging face pipeline params

Reply:
> if this param is not a uri i am confused what 'path' would look like here besides the id itself.

For this model on the hub https://huggingface.co/microsoft/phi-1_5, the model argument would be `microsoft/phi-1_5`

Reply:
Well the goal with keeping this model is that it can take both model-id and the actual loaded model. The actual loaded model works fine for AutoClasses that I tried with but with bigger models like llama2, it threw an error (but worked with passing model-id). 

This was the whole point of updating the doc. 

But I guess this would be temporary until we find out why it doesn't take the loaded model as param.

Reply:
ok, this LGTM with example usage.

Reply:
> Well the goal with keeping this as "model" is that it can take both "model-id" and the "actual loaded model"

what would passing an 'actual loaded model' look like? is this supported too?

Reply:
it is like defining the model with config first and then passing it like this:
```
model_id = 'meta-llama/Llama-2-7b-hf'

model_config = transformers.AutoConfig.from_pretrained(
    model_id,
)

model = transformers.AutoModelForCausalLM.from_pretrained(
    model_id,
    config=model_config,
)

model_handler = HuggingFacePipelineModelHandler(
    task='text-generation',
    model = model,
    load_pipeline_args = {'framework' : 'pt'},
    inference_args = {'max_length' : 200}
)
```

Reply:
ok. so the docstring currently intentionally doesn't document this option or allow it in  a type, and we plan it for the future?